src/Util/TestListenerTrait.php

```diff
@@ -27,6 +27,11 @@ public function startTestSuite($mainSuite)
         $SkippedTestError = class_exists('PHPUnit\Framework\SkippedTestError') ? 'PHPUnit\Framework\SkippedTestError' : 'PHPUnit_Framework_SkippedTestError';
 
         foreach ($mainSuite->tests() as $suite) {
+            // We do not want to test the uuid polyfill on PHP < 7
+            // Since this script tries to load the SUT, we must skip it
+            if ('Symfony\Polyfill\Tests\Uuid\UuidTest' === $suite->getName() && PHP_VERSION_ID < 70000) {
```

Comment:
can we implemenent that in a more generic way by looking at the `@requires php` annotations on the test class instead ?

Reply:
btw, why do we need to make that filtering here ? Is it only because the polyfill implementation uses PHP 7 typehints ? If yes, it might make more sense to drop them (especially given that the polyfilled functions don't have them).

Reply:
> can we implemenent that in a more generic way by looking at the @requires php annotations on the test class instead ?

We could, but there is already some harcoded check in this file. More over we don't have an annotation reader in this project.But maybe phpunit expose that information. If it's the case, I will update the code. If not, I prefer to keep something simple

> Is it only because the polyfill implementation uses PHP 7 typehints ? 

yes I guess

> If yes, it might make more sense to drop them (especially given that the polyfilled functions don't have them).

If I do that, I will have to manually check the argument type by hand to have the same behavior than the ext.

Reply:
does the ext actually throw TypeError or warnings ?

Reply:
It seems it triggers warning: 
```
$ php -a
php > uuid_create('a');
PHP Warning:  uuid_create() expects parameter 1 to be int, string given in php shell code on line 1

Warning: uuid_create() expects parameter 1 to be int, string given in php shell code on line 1
```

Reply:
then the polyfill must trigger warnings too

Reply:
OK, I used PHPUnit groups to fixed this issue. I was not able to retrieve requirements :(

Reply:
well, if you change the polyfill to properly trigger warnings rather than TypeErrors, you might not need this at all anymore (as it will become compatible with PHP 5.x too)